packages/ptokens-assets-evm/src/ptokens-evm-asset-builder.ts

```diff
@@ -30,6 +30,37 @@ export class pTokensEvmAssetBuilder extends pTokensAssetBuilder {
       method: 'hub',
       abi: factoryAbi,
     })
+
+    if (!this._assetInfo.assetTokenAddress) {
```

Comment:
I would do these checks:
if !this._assetInfo.assetTokenAddress -> ok calculate the address.
else if this._assetInfo.assetTokenAddress != this._assetInfo.underlyingTokenAddress -> calculate again ptoken address and check against assetTokenAddress; in case of mismatch throw an error "Invalid pToken address"

Reply:
good idea!